operator/cmd/root.go

```diff
@@ -512,14 +512,18 @@ func (legacy *legacyOnLeader) onStart(_ hive.HookContext) error {
 							// k8s service for etcd. As soon the k8s caches are
 							// synced, this hijack will stop happening.
 							sc := k8s.NewServiceCache(nil)
-							slimSvcObj := k8s.ConvertToK8sService(k8sSvc)
-							slimSvc := k8s.ObjToV1Services(slimSvcObj)
-							if slimSvc == nil {
-								// This will never happen but still log it
-								scopedLog.Warnf("BUG: invalid k8s service: %s", slimSvcObj)
+							slimSvcObj, err := k8s.TransformToK8sService(k8sSvc)
+							if err != nil {
+								scopedLog.Errorf("Failed to transform k8s service %s/%s", k8sSvc.Namespace, k8sSvc.Name)
```

Comment:
There should be some relevant logfields in `pkg/logging/logfields` that we can use here without making the log msg dynamic. This enables the use of structured logging.

```suggestion
								scopedLog.WithError(err).WithFields({
								  ...
								}.)Error("Failed to transform k8s service")
```